mantidproject/mantid
# Indirect - Diffraction Reduction - Incorrect summing of files

Currently, the way that the Sum Files option in the Indirect Diffraction Reduction interface functions is incorrect. The way it has previously been conceptualised is that the list of sample and vanadium runs are summed each into a single workspace, and the reduction is then done with the created workspaces.

The way that this should work, is that when the Sum Files option (now the Sum Sample Files option - to avoid confusion) is checked, vanadium input will be left unaffected, however sample input will be affected in the following way:

- Sample ranges supplied as A-B where A and B are run numbers and A <= B, will sum all sample runs from A to B, inclusively, for the selected instrument.
- Sample sums supplied as A+B+C... where A, B, C... are run numbers, will sum sample runs
- Supplied list should still be comma separated, and each range, sum and single run, a corresponding vanadium file will be required if 'Use Vanadium File' is checked.

**Example**
Sample Files:      26173-26176, 26177
Vanadium Files:  26178, 26179
Where Sum Sample Files is checked
Will result in the sum of sample runs 26173, 26174, 26175, 2176 for the selected instrument corresponding to vanadium file 26178 in the reduction, and sample run 26177 corresponding to vanadium file 26179 in the reduction.

**Note**
Although support for '+' and '-' has been added to IndirectDiffractionReduction, MWRunFiles does not currently support these, and so these won't be allowed within the interface.

**To test:**
1. Ensure 'Use Data Archive' is enabled.
2. Navigate to Interfaces > Indirect > Diffraction.
3. Ensure the selected instrument is "IRIS"
4. Enter "26173-26176, 26177" into the run numbers field.
5. Ensure the Sum Sample Files checkbox is checked.
6. Ensure the Use Vanadium File checkbox is checked.
7. Enter "26178, 26179" into the Vanadium File field.
8. Run the Algorithm.
9. Ensure two output workspaces are created.



Fixes #20483. 

**Release Notes** 

Change has been added to the release notes.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
